growthstreet_zoo/zoo.py

```diff
@@ -67,25 +61,16 @@ def add_animal(self, animal):
         self.predators_eat_prey()
 
     def predators_eat_prey(self):
-        """Given animals in a cage. Check to see if any of them should be eaten."""
+        """When a  new animal enters a cage, check whether any animals in the cage should be eaten and eat them."""
```

Comment:
As per our conversation before, I still think this code needs a better name, and I don't think the algorithm of using permutations is quite right - seems too complicated.  I think just looping through a list of the alive animals in the cage is a bit simpler, what do you think?